internal/core/data/event.go

```diff
@@ -124,7 +127,35 @@ func addNewEvent(e contract.Event, ctx context.Context) (string, error) {
 	return e.ID, nil
 }
 
-func updateEvent(from contract.Event, ctx context.Context) error {
+func validateEvent(e contract.Event, ctx context.Context) error {
+	err := checkDevice(e.Device, ctx)
```

Comment:
This function is only used by tests, not by actual business logic. Needs to be removed. Looks like the tests also only exist to exercise this function, so they can also conceivably be removed.
- TestAddEventWithValidationValueDescriptorExistsAndIsInvalid
- TestAddEventWithValidationValueDescriptorNotFound
- TestEventValidationValueDescriptorDBError